### ISSUE TITLE
0.22rc breaks extending UnitRegistry Unit

### ISSUE THREAD

**Issue by zobac** [hgrecco/pint#1781]
Previously [you helped me](https://github.com/hgrecco/pint-pandas/issues/145) improve my implementation to extending Quantity and Unit and UnitRegistry. The new 0.22rc breaks this implementation.

```
class CMTRegistry(pint.UnitRegistry):
    _quantity_class = CMTQuantity
    _unit_class = CMTUnit
```

However, once the I set up the UnitRegistry thusly:

```
>>>UREG = CMTRegistry(str(REG_PATH), fmt_locale='sys')
>>>pint.set_application_registry(UREG)
>>>pint.UnitRegistry = lambda : UREG
```

The registry is still using pint.Unit

```
>>>type(UREG.deltaC)
<class 'pint.Unit'>
```

**Comment by hgrecco** [comment]
Yes, this is intentional (not breaking things, but changing the way extending quantities work). The way we were doing this before is incompatible with static analysis and therefore a problem to incorporate typing and autocompletion seamlessly. 

I am working to improve the docs, I would appreciate your feedback. Take a look here:

https://pint.readthedocs.io/en/latest/advanced/custom-registry-class.html

(the docs are building but should be finished soon)

**Comment by hgrecco** [comment]
Thanks! I decided not to export at the top level that symbol and forgot to write it properly in the docs.

PS.- by the way, if it is an open source package checkout this: https://github.com/hgrecco/pint/blob/master/downstream_status.md

**Comment by hgrecco** [comment]
Just released 0.21.1 and 0.22
